index.js

```diff
@@ -105,5 +94,22 @@ function wrapFunction(config, req, res, cb) {
       }
       modules.install(ctx);
       return cb(null, ctx);
+    })
+    .then(function (decryptedToken) {
+      let validAudience = authentication.verifyAudience(decryptedToken);
+      if (!validAudience) {
+        throw new authentication.WrongAudienceError(`Token's audience ${decryptedToken.aud} did not match any for this function.`);
+      }
+      ctx.originalToken = token;
+      ctx.token = decryptedToken;
+      modules.install(ctx);
+      cb(null, ctx);
+    })
+    .catch( function(e) {
+      res.status(500).send({
```

Comment:
Here we should have a `console.log(e)` so that we can see the true error in the logs